Mage/src/main/java/mage/abilities/effects/common/continuous/GainAbilityAttachedEffect.java

```diff
@@ -96,7 +102,11 @@ public boolean apply(Game game, Ability source) {
     private void setText() {
         StringBuilder sb = new StringBuilder();
         sb.append(attachmentType.verb());
-        sb.append(" creature ");
+        if (targetObjectName!=null){
+            sb.append(" " + targetObjectName + " ");
```

Comment:
1. Set default `targetObjectName` as `creature` in constructor (so devs can see default values and you don't need null checks).

2. Possible missing changes in `setText`?
* `sb.append(ability.getRule("this creature"));` -- must use `targetObjectName`?

Reply:
Thanks for the feedback! If I understood correctly, I believe I've made all the suggested changes in commit a6a2511.